src/shogun/lib/Map.h

```diff
@@ -441,6 +440,15 @@ IGNORE_IN_CLASSLIST template<class K, class T> class CMap: public CSGObject
 	}
 
 protected:
+	/** Get the hash of a given key
+	 * @param key a given key
+	 * @return hash of the key
+	 */
+	virtual uint32_t get_hash_value(const K& key)
+	{
+		return CHash::MurmurHash3((uint8_t*)(&key), sizeof(key), 0xDEADBEEF);
```

Comment:
A comment on why we are using this value for the third argument would be helpful.